### PR TITLE
Expose Begin/End methods on FileStream

### PR DESCRIPTION
**Pull request by JeremyKuhne**
Also add [Obsolete] to the Handle property

#11373, #11127

@stephentoub, @ericstj

### CODE REVIEW

**Comment by stephentoub** [comment]
The source changes look good.

What about tests?

File: src/System.IO.FileSystem/src/System.IO.FileSystem.csproj [hunk 1]
```diff
@@ -89,6 +89,7 @@
   <!-- .NET Standard 1.7 -->
   <ItemGroup Condition="'$(PackageTargetFramework)' == 'netstandard1.7' and '$(TargetGroup)' != 'net46' and '$(TargetGroup)' != 'net463'" >
     <Compile Include="System\IO\FileStreamBase.NetStandard17.cs" />
+    <Compile Include="System\IO\FileStream.NetStandard17.cs" />
```
Comment (jkotas) [inline src/System.IO.FileSystem/src/System.IO.FileSystem.csproj]: Why is this in a separate file? I would expect that we will just redist the pre-netstandard1.7 binaries; and just build the netstandard17 version from the sources.

**Comment by stephentoub** [comment]
> I would expect that we will just redist the pre-netstandard1.7 binaries; and just build the netstandard17 version from the sources.

There is an ongoing debate between @ericstj and @weshaggard about this. We are currently inconsistent in this regard across projects in the repo.

**Comment by JeremyKuhne** [comment]
@stephentoub 
> What about tests?

I have a separate issue for that I'm working on. Wanted to get the functionality flowing.

**Comment by JeremyKuhne** [comment]
@dotnet-bot test  Innerloop Windows_NT Debug Build and Test

**Comment by JeremyKuhne** [comment]
@dotnet-bot test Innerloop CentOS7.1 Release Build and Test

**Comment by JeremyKuhne** [comment]
@dotnet-bot test Innerloop CentOS7.1 Debug Build and Test

**Comment by JeremyKuhne** [comment]
@dotnet-bot test Innerloop CentOS7.1 Debug Build and Test

**Pull request merged**